include/fmt/core.h

```diff
@@ -132,6 +132,14 @@
 #  define FMT_NORETURN
 #endif
 
+#ifndef FMT_MAYBE_UNUSED
+#  if (__cplusplus >= 201703L)
```

Comment:
I suggest replacing this check with `FMT_HAS_CPP_ATTRIBUTE(maybe_unused)`.

Reply:
Updated to use FMT_HAS_CPP_ATTRIBUTE as requested.

Reply:
> Perhaps a more general approach, to test both language version and compiler support within that version

Good idea.

Reply:
Great, thanks - I've added a FMT_HAS_CPP14_ATTRIBUTE to cover FMT_DEPRECATED as well so all of the version+attribute checks now use this approach.